tests/etl/test_perf_data_load.py

```diff
@@ -0,0 +1,221 @@
+                          suite.get('lowerIsBetter', True),
+                          suite.get('extraOptions'),
+                          suite.get('unit'),
+                          perf_push.time)
+        _verify_datum(suite['name'], '', suite['value'], perf_push.time)
+        for subtest in suite['subtests']:
+            _verify_signature(test_repository.name,
+                              perf_datum['framework']['name'],
+                              suite['name'],
+                              subtest['name'],
+                              'my_option_hash',
+                              'my_platform',
+                              subtest.get('lowerIsBetter', True),
+                              suite.get('extraOptions'),
+                              suite.get('unit'),
+                              perf_push.time)
+            _verify_datum(suite['name'], subtest['name'], subtest['value'],
+                          perf_push.time)
+
+
+def test_hash_remains_unchanged(test_repository, perf_job, datum):
+    perf_datum, submit_datum = _prepare_test_data(datum)
+    store_performance_artifact(perf_job, submit_datum)
+
+    summary_signature = PerformanceSignature.objects.get(
+        suite=perf_datum['suites'][0]['name'], test='')
+    # Ensure we don't inadvertently change the way we generate signature hashes.
+    assert summary_signature.signature_hash == 'f451f0c9000a7f99e5dc2f05792bfdb0e11d0cac'
+    subtest_signatures = PerformanceSignature.objects.filter(
+        parent_signature=summary_signature).values_list('signature_hash', flat=True)
+    assert len(subtest_signatures) == 3
+
+
+def test_timestamp_can_be_updated(test_repository, perf_job, later_perf_push, generic_reference_data, datum):
+    perf_datum, submit_datum = _prepare_test_data(datum)
+    store_performance_artifact(perf_job, submit_datum)
+
+    # send another datum, a little later, verify that signature is changed accordingly
+    later_job = create_generic_job('lateguid', test_repository,
+                                   later_perf_push.id, generic_reference_data)
+    store_performance_artifact(later_job, submit_datum)
+
+    signature = PerformanceSignature.objects.get(
+        suite=perf_datum['suites'][0]['name'],
+        test=perf_datum['suites'][0]['subtests'][0]['name'])
+    assert signature.last_updated == later_perf_push.time
+
+
+def test_measurement_unit_can_be_updated(test_repository, later_perf_push, perf_job, generic_reference_data, datum, datum_with_new_unit):
+    perf_datum, submit_datum = _prepare_test_data(datum)
+    store_performance_artifact(perf_job, submit_datum)
+
+    updated_perf_datum, updated_submit_datum = _prepare_test_data(datum_with_new_unit)
+    later_job = create_generic_job('lateguid', test_repository,
+                                   later_perf_push.id, generic_reference_data)
+    store_performance_artifact(later_job, updated_submit_datum)
+
+    summary_signature = PerformanceSignature.objects.get(
```

Comment:
We need to test the measurement unit for the children signatures as well.

Reply:
Ok.